core/src/main/resources/hudson/security/csrf/CrumbFilter/retry.jelly

```diff
@@ -0,0 +1,42 @@
+<!--
+The MIT License
+
+Copyright (c) 2017, Daniel Beck
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+    <st:statusCode value="405"/>
+    <l:layout norefresh="true" title="${%Method Not Allowed}">
+
+        <l:main-panel>
+            <h1>${%This URL requires POST}</h1>
+            <p>
+                ${%blurb}
+            </p>
+            <p><tt>${requestURL}</tt></p>
+            <p><strong>${%warning}</strong></p>
```

Comment:
Can we check the `Referer` _[sic]_?

Reply:
Check how? Or do you mean we should just print the referer as well, in addition to the URL?